common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java

```diff
@@ -147,13 +101,48 @@ public void run() {
                         }
                     });
                 } else {
-                    final boolean finalResult = result;
-                    handler.post(new Runnable() {
-                        @Override
-                        public void run() {
-                            command.getCallback().onTaskCompleted(finalResult);
+                    if(result != null && result instanceof AcquireTokenResult){
+                        AcquireTokenResult localResult = (AcquireTokenResult)result;
+                        if(localResult.getSucceeded()){
+                            final ILocalAuthenticationResult authenticationResult = localResult.getLocalAuthenticationResult();
+                            handler.post(new Runnable() {
+                                @Override
+                                public void run() {
+                                    command.getCallback().onTaskCompleted(authenticationResult);
+                                }
+                            });
+                        }else{
+                            //Get MsalException from Authorization and/or Token Error Response
+                            baseException = ExceptionAdapter.exceptionFromAcquireTokenResult(localResult);
+                            final BaseException finalException = baseException;
+
+                            if (finalException instanceof UserCancelException) {
+                                //Post Cancel
+                                handler.post(new Runnable() {
+                                    @Override
+                                    public void run() {
+                                        command.getCallback().onCancel();
+                                    }
+                                });
+                            } else {
+                                handler.post(new Runnable() {
+                                    @Override
+                                    public void run() {
+                                        command.getCallback().onError(finalException);
+                                    }
+                                });
+                            }
                         }
-                    });
+                    }else{
+                        final Object returnResult = result;
```

Comment:
should we null check for `returnResult` here ?, Also may be worth adding a comment that we get into the else loop here for `LoadAccountCommand` and `RemoveAccountCommand`